qutebrowser/browser/qutescheme.py

```diff
@@ -158,6 +161,59 @@ def qute_bookmarks(_url):
     return 'text/html', html
 
 
+@add_handler('history')
+def qute_history(url):
+    """Handler for qute:history. Display history."""
+    # Get current date from query parameter, if not given choose today.
+    curr_date = datetime.date.today()
+    try:
+        query_date = QUrlQuery(url).queryItemValue("date")
+        if query_date:
+            curr_date = datetime.datetime.strptime(query_date, "%Y-%m-%d")
+            curr_date = curr_date.date()
+    except ValueError:
+        log.misc.debug("Invalid date passed to qute:history: " + query_date)
+
+    one_day = datetime.timedelta(days=1)
+    next_date = curr_date + one_day
+    prev_date = curr_date - one_day
+
+    def history_iter():
+        for item in objreg.get('web-history').history_dict.values():
```

Comment:
> Hmm, somehow I misread @The-Compiler's post as proposing to just reverse the history before iterating. So just discard all my previous comments here...

That essentially was my proposal, because then it's possible to abort iterating as soon as we encounter an old value. Hopefully my patch below makes things clear. If not, please ask!

> f is_reversed:
        today_history.append(...)
    else:
        today_history.insert(0, ...)
```

> Also, I think that even `history = reversed(list(objreg.get('web-history').history_dict.values()))` for older python versions would be a lot faster than iterating over the whole history.

I don't think so - that does iterate over the whole history, make a copy of it, and then reverse that copy.